frontend/packages/dev-console/src/utils/pipeline-augment.ts

```diff
@@ -396,9 +397,21 @@ export const getRunStatusColor = (status: string): StatusMessage => {
 export const truncateName = (name: string, length: number): string =>
   name.length < length ? name : `${name.slice(0, length - 1)}...`;
 
+/**
+ * Returns the pipeline with the spec at the time of pipeline run execution.
+ * @param pipeline
+ * @param pipelineRun
+ */
+export const getExecutedPipeline = (pipeline: Pipeline, pipelineRun: PipelineRun): Pipeline => {
+  return {
+    ...pipeline,
+    ...(pipelineRun?.status?.pipelineSpec && { spec: pipelineRun?.status?.pipelineSpec }),
+  };
+};
```

Comment:
I think we want to do two things here:

1. Remove the dependancy on Pipeline and trim it out along all the call paths (I just did this and it's quite a bit of changes and adjustments, no references to Pipeline should exist in the call paths)
2. Rename it to something like (don't have to take this if you have something better) `getPipelineFromPipelineRun`

Tests will need to be updated naturally when this happens, since `getTaskStatus` heavily uses Pipeline. We may even need to update the test data to add `pipelineSpec` (which should be a mirror - maybe break that out to a var? - of the pipeline.spec used in the same data structure)

Reply:
Removed the pipeline dependency in the call paths, updated test data and tests as well.